<a href="https://colab.research.google.com/github/SureshTakasi/DATA602-ML-UMBC/blob/main/Amazon_Product_Review_Sentiment_Analysis_and_Star_Rating_Prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Import libraries

In [14]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import time
import sys
import os
import gzip

"""
We will ignore FutureWarning and DeprecationWarning
"""
import warnings
warnings.filterwarnings("ignore", category=FutureWarning)
warnings.filterwarnings("ignore", category=DeprecationWarning)

"""
We will ignore warnings
"""
warnings.filterwarnings("ignore")

if not sys.warnoptions:
    warnings.simplefilter("ignore")
program_start_time=time.time()

Download dataset

In [15]:
!wget https://s3.amazonaws.com/amazon-reviews-pds/tsv/amazon_reviews_us_Mobile_Electronics_v1_00.tsv.gz

--2020-10-08 14:53:27--  https://s3.amazonaws.com/amazon-reviews-pds/tsv/amazon_reviews_us_Mobile_Electronics_v1_00.tsv.gz
Resolving s3.amazonaws.com (s3.amazonaws.com)... 52.216.184.93
Connecting to s3.amazonaws.com (s3.amazonaws.com)|52.216.184.93|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 22870508 (22M) [application/x-gzip]
Saving to: ‘amazon_reviews_us_Mobile_Electronics_v1_00.tsv.gz.1’

amazon_reviews_us_M 100%[===================>]  21.81M  33.0MB/s    in 0.7s    

2020-10-08 14:53:28 (33.0 MB/s) - ‘amazon_reviews_us_Mobile_Electronics_v1_00.tsv.gz.1’ saved [22870508/22870508]



Open dataset and save as dataframe

In [16]:
import gzip
with gzip.open('amazon_reviews_us_Mobile_Electronics_v1_00.tsv.gz') as f:
    data = pd.read_csv(f, sep='\t', error_bad_lines=False)

data.head()

b'Skipping line 35246: expected 15 fields, saw 22\n'
b'Skipping line 87073: expected 15 fields, saw 22\n'


marketplace  ...  review_date
0          US  ...   2015-08-31
1          US  ...   2015-08-31
2          US  ...   2015-08-31
3          US  ...   2015-08-31
4          US  ...   2015-08-31

[5 rows x 15 columns]

#### Data Preprocessing

Check info of dataset

In [17]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 104852 entries, 0 to 104851
Data columns (total 15 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   marketplace        104852 non-null  object 
 1   customer_id        104852 non-null  int64  
 2   review_id          104852 non-null  object 
 3   product_id         104852 non-null  object 
 4   product_parent     104852 non-null  int64  
 5   product_title      104852 non-null  object 
 6   product_category   104852 non-null  object 
 7   star_rating        104850 non-null  float64
 8   helpful_votes      104850 non-null  float64
 9   total_votes        104850 non-null  float64
 10  vine               104850 non-null  object 
 11  verified_purchase  104850 non-null  object 
 12  review_headline    104848 non-null  object 
 13  review_body        104849 non-null  object 
 14  review_date        104850 non-null  object 
dtypes: float64(3), int64(2), object(10)
memory usage: 1

Check shape of data

In [18]:
data.shape

(104852, 15)

Describe the data

In [19]:
data.describe()

customer_id  product_parent  ...  helpful_votes    total_votes
count  1.048520e+05    1.048520e+05  ...  104850.000000  104850.000000
mean   2.793783e+07    5.015196e+08  ...       1.243996       1.615393
std    1.508722e+07    2.871661e+08  ...       7.070387       7.909896
min    1.007100e+04    5.352400e+04  ...       0.000000       0.000000
25%    1.471402e+07    2.593731e+08  ...       0.000000       0.000000
50%    2.650357e+07    4.937289e+08  ...       0.000000       0.000000
75%    4.223551e+07    7.440083e+08  ...       1.000000       1.000000
max    5.309657e+07    9.999508e+08  ...     769.000000     791.000000

[8 rows x 5 columns]

Convert type of columns star_rating, helpful_votes and total_votes from float to int

In [20]:
cols = ['star_rating','helpful_votes', 'total_votes']
data[cols] = data[cols].astype('Int64')
data['star_rating'].head()

0    5
1    5
2    5
3    4
4    2
Name: star_rating, dtype: Int64

Check null values

In [21]:
data.isnull().sum()

marketplace          0
customer_id          0
review_id            0
product_id           0
product_parent       0
product_title        0
product_category     0
star_rating          2
helpful_votes        2
total_votes          2
vine                 2
verified_purchase    2
review_headline      4
review_body          3
review_date          2
dtype: int64

Check where the values are null

In [22]:
null_data = data[data.isnull().any(axis=1)]
null_data

marketplace  ...  review_date
20969          US  ...   2014-11-11
60183          US  ...          NaN
62888          US  ...          NaN
66724          US  ...   2012-11-24
69533          US  ...   2012-09-05

[5 rows x 15 columns]

We can see two rows with ids 60183 and 62888 have no values important for prediction. We can remove those values